src/librbd/mirror/snapshot/UnlinkPeerRequest.cc

```diff
@@ -187,8 +187,9 @@ void UnlinkPeerRequest<I>::remove_snapshot() {
     &snap_namespace);
   ceph_assert(info);
 
-  if (info->mirror_peer_uuids.size() > 1 ||
-      info->mirror_peer_uuids.count(m_mirror_peer_uuid) == 0) {
+  if ((info->mirror_peer_uuids.size() > 1 ||
+       info->mirror_peer_uuids.count(m_mirror_peer_uuid) == 0) &&
+      (!info->mirror_peer_uuids.empty() || !m_newer_mirror_snapshots)) {
```

Comment:
Just thinking, could it be made more readable? E.g. it seems to be equivalent of the below:
```
       info->mirror_peer_uuids.erase(m_mirror_peer_uuid);
       if (!info->mirror_peer_uuids.empty() || !m_newer_mirror_snapshots) {
         // skip
       }
```

Reply:
I think the matching simplification would be:

```
auto removed_count = info->mirror_peer_uuids.erase(m_mirror_peer_uuid);
if (!info->mirror_peer_uuids.empty() ||
   (!m_newer_mirror_snapshots && removed_count == 0))) {
```

but your version should also be correct since even if we don't remove our peer uuid, we probably shouldn't remove the most recent mirror snapshot.

Reply:
... but now that I think about it, we don't want to tweak the data structure if we aren't going to remove it, so `info` would need to be a copy as well.

Reply:
I was thinking it would be safe to tweak (remove a peer from) `snapshot_namespace`, because it could be only used for snap_remove operation and it would not affect this?

Anyway I am fine with the current version.